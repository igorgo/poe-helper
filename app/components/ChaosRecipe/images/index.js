import amulet from './amulet.png';
import belt from './belt.png';
import body from './body.png';
import boot from './boot.png';
import glove from './glove.png';
import helmet from './helmet.png';
import ring from './ring.png';
import weapon from './weapon.png';

export default {
  amulet,
  belt,
  body,
  boot,
  glove,
  helmet,
  ring,
  weapon
};
