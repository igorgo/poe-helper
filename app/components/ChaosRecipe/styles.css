.contentContainer {
  height: calc(100% - 30px);
}

.container {
  width: 360px;
}

.buttonsContainer {
  height: 30px;
  padding: 10px 0;
}

.gridContainer {
  height: calc(100% - 30px);
}

.item {
  display: inline-block;
}
