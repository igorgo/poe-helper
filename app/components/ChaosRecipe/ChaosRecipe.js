/* eslint-disable react/no-array-index-key,jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */
// @flow

import React from 'react';
import styles from './styles.css';
import pics from './images';
import type {
  ChaosRecipeItemType,
  ChaosRecipeStore
} from '../../reducers/types';
import routes from '../../constants/routes';
import NavMenu from '../NavMenu/NavMenu';

export const rows = {
  weapon: {
    length: 12,
    height: 3,
    width: 1
  },
  helmet: {
    length: 6,
    height: 2,
    width: 2
  },
  body: {
    length: 6,
    height: 3,
    width: 2
  },
  glove: {
    length: 6,
    height: 2,
    width: 2
  },
  boot: {
    length: 6,
    height: 2,
    width: 2
  },
  ring: {
    length: 12,
    height: 1,
    width: 1
  },
  amulet: {
    length: 6,
    height: 1,
    width: 1
  },
  belt: {
    length: 6,
    height: 1,
    width: 2
  }
};

type PoeItemProps = {
  pic: string,
  width: number,
  height: number,
  value: boolean,
  id: string
};

const PoeItem = (props: PoeItemProps) => {
  const { pic, width, height, value, id } = props;
  return (
    <img
      id={id}
      className="col-auto"
      alt={pic}
      src={pics[pic]}
      width={width * 30}
      height={height * 30}
      style={{
        opacity: value ? 1 : 0.2
      }}
    />
  );
};

type PoeItemRowProps = {
  type: ChaosRecipeItemType,
  length: number,
  height: number,
  width: number,
  value: number,
  onIncrement: (itemType: ChaosRecipeItemType) => void,
  onDecrement: (itemType: ChaosRecipeItemType) => void
};

export const PoeItemRow = (props: PoeItemRowProps) => {
  const {
    length = 6,
    height = 2,
    width = 2,
    type,
    value,
    onIncrement,
    onDecrement
  } = props;
  const handleLeftClick = event => {
    event.stopPropagation();
    if (onIncrement) {
      onIncrement(type);
    }
  };

  const handleRightClick = event => {
    event.preventDefault();
    event.stopPropagation();
    if (onDecrement) {
      onDecrement(type);
    }
  };
  return (
    <div
      className="row justify-around"
      onClick={handleLeftClick}
      onContextMenu={handleRightClick}
    >
      {[...Array(length)].map((item, idx) => (
        <PoeItem
          key={`${type}-${idx}`}
          id={`${type}${idx}`}
          pic={type}
          width={width}
          height={height}
          value={value > idx}
        />
      ))}
    </div>
  );
};

type ChaosRecipeProps = {
  counts: ChaosRecipeStore,
  increment: (itemType: ChaosRecipeItemType) => void,
  decrement: (itemType: ChaosRecipeItemType) => void,
  reset: () => void,
  minusSet: () => void
};

const ChaosRecipeHelper = (props: ChaosRecipeProps) => {
  const { counts, increment, decrement, reset, minusSet } = props;
  return (
    <div className="vp-height full-width non-selectable column">
      <div
        className={`row justify-center items-center ${styles.contentContainer}`}
      >
        <div className={`${styles.container} col-auto`}>
          <div
            className={`${styles.buttonsContainer} row justify-around items-center`}
          >
            <button
              type="button"
              className={`${styles.btn} col-auto`}
              onClick={reset}
            >
              Reset
            </button>
            <button
              type="button"
              className={`${styles.btn} col-auto`}
              onClick={minusSet}
            >
              Minus set
            </button>
          </div>
          <div className={`${styles.gridContainer} column`}>
            {Object.keys(rows).map(type => (
              <PoeItemRow
                key={type}
                type={type}
                length={rows[type].length}
                height={rows[type].height}
                width={rows[type].width}
                value={counts[type]}
                onDecrement={decrement}
                onIncrement={increment}
              />
            ))}
          </div>
        </div>
      </div>
      <NavMenu exclude={routes.CHAOS_RECIPE} />
    </div>
  );
};

export default ChaosRecipeHelper;
