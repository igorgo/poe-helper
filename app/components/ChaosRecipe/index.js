import ChaosRecipe from './ChaosRecipe';

export default ChaosRecipe;
