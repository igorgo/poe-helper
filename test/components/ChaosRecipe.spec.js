import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ChaosRecipe, {
  PoeItemRow
} from '../../app/components/ChaosRecipe/ChaosRecipe';

Enzyme.configure({ adapter: new Adapter() });

const counts = {
  weapon: 0,
  helmet: 0,
  body: 1,
  glove: 0,
  boot: 0,
  ring: 0,
  amulet: 0,
  belt: 0
};

function setup() {
  const actions = {
    increment: spy(),
    reset: spy(),
    minusSet: spy(),
    decrement: spy()
  };
  const component = shallow(<ChaosRecipe counts={counts} {...actions} />);
  return {
    component,
    actions,
    buttons: component.find('button'),
    bodyRow: component.find('PoeItemRow').at(2)
  };
}

function setupRow() {
  const actions = {
    onIncrement: spy(),
    onDecrement: spy()
  };
  return shallow(
    <PoeItemRow
      type="body"
      length={6}
      height={3}
      width={2}
      value={2}
      {...actions}
    />
  );
}

describe('ChaosRecipe component', () => {
  it('should contains body row with value 1', () => {
    const { bodyRow } = setup();
    console.log(bodyRow.debug());
    console.log(bodyRow.debug());
    expect(bodyRow.props().value).toBe(1);
  });

  it('should call increment on PoeItemRow click', () => {
    const { bodyRow, actions } = setup();
    bodyRow.simulate('increment');
    expect(actions.increment.called).toBe(true);
  });

  it('should call decrement on PoeItemRow click', () => {
    const { bodyRow, actions } = setup();
    bodyRow.simulate('decrement');
    expect(actions.decrement.called).toBe(true);
  });

  it('should first button should call reset', () => {
    const { buttons, actions } = setup();
    buttons.at(0).simulate('click');
    expect(actions.reset.called).toBe(true);
  });

  it('should second button should call minusSet', () => {
    const { buttons, actions } = setup();
    buttons.at(1).simulate('click');
    expect(actions.minusSet.called).toBe(true);
  });

  it('should 2nd items should have value true', () => {
    const component = setupRow();
    expect(
      component
        .find('PoeItem')
        .at(1)
        .props().value
    ).toBe(true);
  });

  it('should 3rd items should have value false', () => {
    const component = setupRow();
    expect(
      component
        .find('PoeItem')
        .at(2)
        .props().value
    ).toBe(false);
  });

  it('should match exact snapshot', () => {
    const { actions } = setup();
    const counter = (
      <div>
        <Router>
          <ChaosRecipe counts={counts} {...actions} />
        </Router>
      </div>
    );
    const tree = renderer.create(counter).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
