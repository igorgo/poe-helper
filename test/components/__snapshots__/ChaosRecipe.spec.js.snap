// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChaosRecipe component should match exact snapshot 1`] = `
<div>
  <div
    className="vp-height full-width non-selectable column"
  >
    <div
      className="row justify-center items-center contentContainer"
    >
      <div
        className="container col-auto"
      >
        <div
          className="buttonsContainer row justify-around items-center"
        >
          <button
            className="btn col-auto"
            onClick={[Function]}
            type="button"
          >
            Reset
          </button>
          <button
            className="btn col-auto"
            onClick={[Function]}
            type="button"
          >
            Minus set
          </button>
        </div>
        <div
          className="gridContainer column"
        >
          <div
            className="row justify-around"
            onClick={[Function]}
            onContextMenu={[Function]}
          >
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon0"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon1"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon2"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon3"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon4"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon5"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon6"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon7"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon8"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon9"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon10"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="weapon"
              className="col-auto"
              height={90}
              id="weapon11"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
          </div>
          <div
            className="row justify-around"
            onClick={[Function]}
            onContextMenu={[Function]}
          >
            <img
              alt="helmet"
              className="col-auto"
              height={60}
              id="helmet0"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="helmet"
              className="col-auto"
              height={60}
              id="helmet1"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="helmet"
              className="col-auto"
              height={60}
              id="helmet2"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="helmet"
              className="col-auto"
              height={60}
              id="helmet3"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="helmet"
              className="col-auto"
              height={60}
              id="helmet4"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="helmet"
              className="col-auto"
              height={60}
              id="helmet5"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
          </div>
          <div
            className="row justify-around"
            onClick={[Function]}
            onContextMenu={[Function]}
          >
            <img
              alt="body"
              className="col-auto"
              height={90}
              id="body0"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 1,
                }
              }
              width={60}
            />
            <img
              alt="body"
              className="col-auto"
              height={90}
              id="body1"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="body"
              className="col-auto"
              height={90}
              id="body2"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="body"
              className="col-auto"
              height={90}
              id="body3"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="body"
              className="col-auto"
              height={90}
              id="body4"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="body"
              className="col-auto"
              height={90}
              id="body5"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
          </div>
          <div
            className="row justify-around"
            onClick={[Function]}
            onContextMenu={[Function]}
          >
            <img
              alt="glove"
              className="col-auto"
              height={60}
              id="glove0"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="glove"
              className="col-auto"
              height={60}
              id="glove1"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="glove"
              className="col-auto"
              height={60}
              id="glove2"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="glove"
              className="col-auto"
              height={60}
              id="glove3"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="glove"
              className="col-auto"
              height={60}
              id="glove4"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="glove"
              className="col-auto"
              height={60}
              id="glove5"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
          </div>
          <div
            className="row justify-around"
            onClick={[Function]}
            onContextMenu={[Function]}
          >
            <img
              alt="boot"
              className="col-auto"
              height={60}
              id="boot0"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="boot"
              className="col-auto"
              height={60}
              id="boot1"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="boot"
              className="col-auto"
              height={60}
              id="boot2"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="boot"
              className="col-auto"
              height={60}
              id="boot3"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="boot"
              className="col-auto"
              height={60}
              id="boot4"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="boot"
              className="col-auto"
              height={60}
              id="boot5"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
          </div>
          <div
            className="row justify-around"
            onClick={[Function]}
            onContextMenu={[Function]}
          >
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring0"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring1"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring2"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring3"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring4"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring5"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring6"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring7"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring8"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring9"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring10"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="ring"
              className="col-auto"
              height={30}
              id="ring11"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
          </div>
          <div
            className="row justify-around"
            onClick={[Function]}
            onContextMenu={[Function]}
          >
            <img
              alt="amulet"
              className="col-auto"
              height={30}
              id="amulet0"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="amulet"
              className="col-auto"
              height={30}
              id="amulet1"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="amulet"
              className="col-auto"
              height={30}
              id="amulet2"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="amulet"
              className="col-auto"
              height={30}
              id="amulet3"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="amulet"
              className="col-auto"
              height={30}
              id="amulet4"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
            <img
              alt="amulet"
              className="col-auto"
              height={30}
              id="amulet5"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={30}
            />
          </div>
          <div
            className="row justify-around"
            onClick={[Function]}
            onContextMenu={[Function]}
          >
            <img
              alt="belt"
              className="col-auto"
              height={30}
              id="belt0"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="belt"
              className="col-auto"
              height={30}
              id="belt1"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="belt"
              className="col-auto"
              height={30}
              id="belt2"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="belt"
              className="col-auto"
              height={30}
              id="belt3"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="belt"
              className="col-auto"
              height={30}
              id="belt4"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
            <img
              alt="belt"
              className="col-auto"
              height={30}
              id="belt5"
              src="test-file-stub"
              style={
                Object {
                  "opacity": 0.2,
                }
              }
              width={60}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="container row justify-around"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
  </div>
</div>
`;
